bm-persona/Dining/DiningDetailViewController.swift

```diff
@@ -0,0 +1,496 @@
+        card.heightAnchor.constraint(equalToConstant: 200).isActive = true
+        view.layoutSubviews()
+        
+        card.addSubview(clockIcon)
+        card.addSubview(openTimeLabel)
+        card.addSubview(openTag)
+        card.addSubview(chairImage)
+        card.addSubview(capBadge)
+        card.addSubview(faveButton)
+        card.addSubview(diningImage)
+        card.addSubview(phoneIcon)
+        card.addSubview(phoneLabel)
+        card.addSubview(distIcon)
+        card.addSubview(distLabel)
+        card.addSubview(addressIcon)
+        card.addSubview(addressLabel)
+        card.addSubview(nameLabel)
+        
+        clockIcon.bottomAnchor.constraint(equalTo: card.layoutMarginsGuide.bottomAnchor).isActive = true
+        clockIcon.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        clockIcon.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        clockIcon.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        
+        let formatter = DateIntervalFormatter()
+        formatter.dateStyle = .none
+        formatter.timeStyle = .short
+        let date = Date()
+        openTimeLabel.text = "     "
+        if let intervals = diningHall!.weeklyHours?.hoursForWeekday(DayOfWeek.weekday(date)) {
+            openTimeLabel.text = "Closed Today"
+            var nextOpenInterval: DateInterval? = nil
+            for interval in intervals {
+                if interval.contains(date) {
+                    nextOpenInterval = nil
+                    openTimeLabel.text = formatter.string(from: interval.start, to: interval.end)
+                    break
+                } else if date.compare(interval.start) == .orderedAscending {
+                    if nextOpenInterval == nil {
+                        nextOpenInterval = interval
+                    } else if interval.compare(nextOpenInterval!) == .orderedAscending {
+                        nextOpenInterval = interval
+                    }
+                }
+            }
+            if nextOpenInterval != nil {
+                openTimeLabel.text = formatter.string(from: nextOpenInterval!.start, to: nextOpenInterval!.end)
+            }
+        }
+        openTimeLabel.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        openTimeLabel.leftAnchor.constraint(equalTo: clockIcon.rightAnchor, constant: 5).isActive = true
+        
+        if diningHall!.isOpen! {
+            openTag.text = "Open"
+            openTag.backgroundColor = Color.openTag
+        } else {
+            openTag.text = "Closed"
+            openTag.backgroundColor = Color.closedTag
+        }
+        openTag.leftAnchor.constraint(equalTo: openTimeLabel.rightAnchor, constant: kViewMargin).isActive = true
+        openTag.centerYAnchor.constraint(equalTo: clockIcon.centerYAnchor).isActive = true
+        openTag.widthAnchor.constraint(equalToConstant: 50).isActive = true
```

Comment:
I guess for future reference, I'm pretty sure you don't need to specify a width constraint for a TagView unless you wanna.

Reply:
yeah this was because without the width constraint the tags would be too small or disappear if the opening times label is too long. I wanted to constrain everything so that the only thing that could change in size was the label